### PR TITLE
Parse CDS units with Unit 'cds' format.

### PR DESCRIPTION
**Pull request by kyleaoman**
The astropy.units.Unit class has an option 'format' in the constructor which accepts a parameter 'cds' so that units formatted conforming to the CDS standard can be parsed correctly. This change sets the column units of the read table using this feature. Previously, the columns of the read table would have units of type UnrecognizedUnit (basically a glorified string) in many cases - I stumbled on this when trying to read a table with a column in GMsun (giga-solar-masses). I believe this should be fully backwards-compatible from a user point of view.

The following snippet will fail before this change, and succeed after:

```python
from astropy.io import ascii
import astropy.units as u

reader = ascii.get_reader(ascii.Cds, readme='ftp://cdsarc.u-strasbg.fr/pub/cats/J/AJ/152/157/ReadMe')
SPARC = reader.read('ftp://cdsarc.u-strasbg.fr/pub/cats/J/AJ/152/157/table1.dat')
print(SPARC['MHI'].to(u.solMass))
```

### CODE REVIEW

**Comment by astrofrog** [comment]
@kyleaoman - can you add the example you mention as a unit test for the CDS reader?

**Comment by kyleaoman** [comment]
Added a test. I merged changes from astropy/master in my branch a while ago, which I now see from the instructions was a bad idea.

**Comment by kyleaoman** [comment]
The change to the cds unit parsing seems to break some other tests, this will need to be looked into...

**Comment by bsipocz** [comment]
@kyleaoman - If you could rebase that would be the way to get rid of the merge issues. I'm happy to help with it if needed, just ping me on slack.

**Comment by kyleaoman** [comment]
@astrofrog added the test, checked that it fails before and passes after. Also cleaned up a bit with help from @bsipocz . There are a couple of other tests that now fail, which I'm having some trouble diagnosing. If one of you can help, or can ping someone else who can, that would be much appreciated.

**Comment by kyleaoman** [comment]
Seem to have sorted out the issues with failing tests. Looks like all that's missing is a review from @taldcroft.

File: astropy/io/ascii/cds.py [hunk 1]
```diff
@@ -114,9 +116,15 @@ def get_cols(self, lines):
...
                                        match.group('start') or match.group('end'))) - 1
                 col.end = int(match.group('end'))
-                col.unit = match.group('units')
-                if col.unit == '---':
+                unit = match.group('units')
+                if unit == '---':
                     col.unit = None  # "---" is the marker for no unit in CDS table
+                else:
+                    try:
+                        col.unit = Unit(unit, format='cds')
```
Comment (mhvk) [inline astropy/io/ascii/cds.py]: I'm not sure we should fail on the unit not parsing correctly - that forces the user to edit the files they downloaded. Instead, I would suggest to follow the fits example and just add `parse_strict='warn'` or even `parse_strict='silent'` (which means one get `UnrecognizedUnit` after all, but of course only for those units that are, in fact, not recognized).

But best to cc @taldcroft, since he will have a better idea about what the standard is in `io.ascii` (perhaps there is a keyword argument already for it??).

**Comment by kyleaoman** [comment]
I agree that a warning makes more sense than an error if a unit fails to parse. A user can always suppress a warning if they want it to be silent. @taldcroft hopefully this is ready for merge now?

File: astropy/utils/iers/tests/test_iers.py [hunk 2]
```diff
@@ -34,14 +34,14 @@ def test_simple(self):
         iers_tab = iers.IERS.open()
         assert iers.IERS.iers_table is not None
         assert isinstance(iers.IERS.iers_table, QTable)
-        assert iers_tab['UT1_UTC'].unit is u.second
-        assert iers_tab['PM_x'].unit is u.arcsecond
-        assert iers_tab['PM_y'].unit is u.arcsecond
+        assert (iers_tab['UT1_UTC'].unit / u.second).is_unity()
```
Comment (taldcroft) [inline astropy/utils/iers/tests/test_iers.py]: Humm, this is slightly disconcerting.  I see the nice commit description that points to why this change was made.  For example:
```
In [5]: u.s is u.Unit('s')
Out[5]: True

In [6]: u.s is u.Unit('s', format='cds')
Out[6]: False

In [7]: u.s / u.s
Out[7]: Unit(dimensionless)

In [8]: u.s / u.Unit('s', format='cds')
Out[8]: Unit("s / s")
```
I had imagined that the unit `format` implies how to parse the unit string (or combinations thereof), but that the fundamental units coming out were the same thing.  Not so apparently?

So this does imply some risk of breaking code with this PR.  Thoughts @mhvk?

**Comment by taldcroft** [comment]
@kyleaoman - I think for this change it is not necessary to include the two new (large-ish) test files.  You could instead just modify the following one to set a unit of `GMsun` for the `Fit` column.

https://github.com/astropy/astropy/blob/master/astropy/io/ascii/tests/t/cds.dat#L27

You should rebase away those files (basically `git rebase -i ..` and squash away 9701dc93, then re-add your new test using the existing (but modified `cds.dat`).  That way you can keep all the original commits and messages.  Let me know if this isn't clear.

**Comment by taldcroft** [comment]
As noted by the bot, you need to put in a changelog entry for this.  I am labeling this as a bug fix because CDS units should be correctly parsed by the CDS reader, so the entry would go in the bug fix section.  

However, it is not obvious to me that this is fully backwards compatible because of the issue with units, so as it stands this should NOT be backported.

**Comment by taldcroft** [comment]
@kyleaoman - apart from the little technical issues, this looks good, thanks for the contribution!

**Comment by taldcroft** [comment]
@mhvk - OK, issue raised in #7751.  If that can fixed then this PR would truly be back-compatible and could presumably be back-ported (assuming the #7751 fix could be back-ported as well).

**Comment by kyleaoman** [comment]
@taldcroft Alright, think all the requested changes are handled, and everything checks out on the automated tests now.

File: astropy/io/ascii/setup_package.py [hunk 3]
```diff
@@ -85,5 +85,7 @@ def get_package_data():
                                    't/fixed_width_2_line.txt',
                                    't/cds/description/ReadMe',
                                    't/cds/description/table.dat',
+                                   't/cds/units/ReadMe',
```
Comment (taldcroft) [inline astropy/io/ascii/setup_package.py]: This needs to be reverted.

Comment (kyleaoman) [reply astropy/io/ascii/setup_package.py]: Reverted.

**Comment by kyleaoman** [comment]
@taldcroft and done.
